### PR TITLE
Added Publish snapshots to maven via GHA workflow

### PR DESCRIPTION
**Pull request by Angie-Zhang**
*Issue #, if available:*
https://github.com/opensearch-project/index-management/issues/689

*Description of changes:*
Added Publish snapshots to maven via GHA workflow

*CheckList:*
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/index-management/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -75,6 +75,17 @@ buildscript {
         maven { url "https://aws.oss.sonatype.org/content/repositories/snapshots" }
     }
 
+    repositories {
```
Comment (gaiksaya) [inline build.gradle:78]: Hi @Angie-Zhang, this block needs to be under `publishing` block. Please see the issue for example and also I added how to test it. Thanks!

Comment (Angie-Zhang) [reply build.gradle]: Updated. tested in local and it works as expected.

**Comment by Angie-Zhang** [comment]
WhiteSource Security Check  fails due to Vulnerable Library: snakeyaml-1.33.jar. This comes from OSCore.

**Review by gaiksaya — APPROVED** [review]
Added the `PUBLISH_SNAPSHOTS_ROLE` secret.

**Review by lezzago — APPROVED** [review]

**Review by bowenlan-amzn — APPROVED** [review]

**Comment by gaiksaya** [comment]
Hi @Angie-Zhang ,
Just to confirm can you post the snapshots output from testing here? We observed a few glitches in other plugins.
Thanks!

**Comment by Angie-Zhang** [comment]
> Hi @Angie-Zhang , Just to confirm can you post the snapshots output from testing here? We observed a few glitches in other plugins. Thanks!

I replied the results in the issue itself: https://github.com/opensearch-project/index-management/issues/689#issuecomment-1438980573

**Pull request merged**